sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -199,7 +210,8 @@ class CodegenContext {
       case StringType => s"$input.getUTF8String($ordinal)"
       case BinaryType => s"$input.getBinary($ordinal)"
       case CalendarIntervalType => s"$input.getInterval($ordinal)"
-      case t: StructType => s"$input.getStruct($ordinal, ${t.size})"
+      case t: StructType => if (!isColumnarType(input)) { s"$input.getStruct($ordinal, ${t.size})" }
```

Comment:
Right now, the parquet reader does not support nested types (Array, Map, Struct), it's fine to not have this special case in this PR.